uds/database/data_record/formula_data_record.py

```diff
@@ -0,0 +1,82 @@
+"""Definition of FormulaDataRecord."""
+
+from dataclasses import dataclass
+from typing import Callable, Optional, Union
+
+from .abstract_data_record import DataRecordPhysicalValueAlias, DecodedDataRecord
+from .raw_data_record import RawDataRecord
+
+
+@dataclass
+class FormulaRange:
+    min_value: int
+    max_value: int
+
+    def __post_init__(self):
+        if not isinstance(self.min_value, int) or not isinstance(self.max_value, int):
+            raise TypeError("Attributes 'min_value' and 'max_value' must be integers.")
+        if self.min_value > self.max_value:
+            raise ValueError("Attribute 'min_value' must be less than 'max_value'.")
+
+
+class CustomFormulaDataRecord(RawDataRecord):
+    def __init__(
+            self,
+            name: str,
+            length: int,
+            decode_formula: Callable,
+            encode_formula: Callable,
+            formula_range: Optional[FormulaRange] = None
+    ) -> None:
+        super().__init__(name=name, length=length)
+
+        if not callable(decode_formula):
+            raise TypeError("Provided 'decode_formula' is not callable.")
+        if not callable(encode_formula):
+            raise TypeError("Provided 'encode_formula' is not callable.")
+        if formula_range is not None and not isinstance(formula_range, FormulaRange):
+            raise TypeError(
+                "Parameter 'formula_range' must be type of 'FormulaRange', "
+                f"provided type: '{type(formula_range).__name__}'."
+            )
+        self.decode_formula = decode_formula
+        self.encode_formula = encode_formula
+        self.formula_range = formula_range
```

Comment:
if `formula_range` argument is None, then we should set full range (0, (1<<length)-1) here..